### PR TITLE
Enable EFA support

### PR DESCRIPTION
**Pull request by ytsssun**
**Issue number:**

Related #1031 

**Description of changes:**
1. Note: rdma-core will be vended as part of the `release` package in core-kit. PR https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/252.
2. Added setting generator for `metadata.settings.kernel.sysctl."vm/nr_hugepages"` in shared default
3. Added migration for setting generator.

**Testing done:**

Tested migration:

- [x] Test upgrade:

**kernel.sysctl setting**
Before upgrade:
```
[ssm-user@control]$ apiclient get settings.kernel
{
  "settings": {
    "kernel": {
      "lockdown": "none"
    }
  }
}
```
Upgrade:
```
[ssm-user@control]$ apiclient get settings.kernel
{
  "settings": {
    "kernel": {
      "lockdown": "none",
      "sysctl": {
        "vm/nr_hugepages": "3520"
      }
    }
  }
}
```

**setting generator for `settings.kernel.sysctl.vm/nr_hugepages`**:
Before:
```
{
    "settings.network.hostname": "netdog generate-hostname",
    "settings.updates.seed": "bork seed",
    "settings.boot": "/usr/bin/prairiedog generate-boot-settings",
    "settings.updates.targets-base-url": "schnauzer-v2 render --requires 'aws@v1' --requires 'updates@v1(helpers=[tuf-prefix])' --template '{{ tuf-prefix settings.aws.region }}/targets/'",
    "settings.host-containers.admin.user-data": "shibaken generate-admin-userdata",
    "settings.host-containers.control.source": "schnauzer-v2 render --requires 'aws@v1(helpers=[ecr-prefix])' --template '{{ ecr-prefix settings.aws.region }}/bottlerocket-control:v0.7.17'",
    "settings.metrics.send-metrics": "shibaken is-partition --partition aws --partition aws-us-gov",
    "settings.aws.config": "schnauzer-v2 render --requires 'aws@v1(helpers=[aws-config])' --template '{{ aws-config settings.aws.config settings.aws.profile }}'",
    "settings.host-containers.admin.source": "schnauzer-v2 render --requires 'aws@v1(helpers=[ecr-prefix])' --template '{{ ecr-prefix settings.aws.region }}/bottlerocket-admin:v0.11.13'",
    "settings.updates.metadata-base-url": "schnauzer-v2 render --requires 'aws@v1' --requires 'updates@v1(helpers=[metadata-prefix, tuf-prefix])' --template '{{ tuf-prefix settings.aws.region }}{{ metadata-prefix settings.aws.region }}/2020-07-07/{{ os.variant_id }}/{{ os.arch }}/'"
}
```

After upgrades:
```
{
  ...
  "settings.kernel.sysctl.vm/nr_hugepages": "corndog generate-hugepages-setting", <--- New
  ...
}
```
- [x] Test downgrade

Verified that settings remains:
```
[root@admin]# apiclient get settings.kernel
{
  "settings": {
    "kernel": {
      "lockdown": "none",
      "sysctl": {
        "vm/nr_hugepages": "3520"
      }
    }
  }
}
```
Settings generator is removed after downgrade.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Comment by bcressey** [comment]
We should require `rdma-core` over in the core kit's `release` package instead. You can affinitize it to just `aws-*` variants though.

As a general rule, I don't like adding to the set of packages all variants need to list to achieve the baseline functionality level.

**Comment by bcressey** [comment]
Is there a reason the ECS variants were omitted from this list? That should be superseded by the change I suggested, but I'm still curious.

**Comment by ytsssun** [comment]
> We should require `rdma-core` over in the core kit's `release` package instead. You can affinitize it to just `aws-*` variants though.

Got it, it makes sense to put the `rdma-core` under `release` package. I could use a pointer on how to affinitize it to just `aws-*` though.



> Is there a reason the ECS variants were omitted from this list? That should be superseded by the change I suggested, but I'm still curious.

Mainly because ECS does not officially support EFA? However, I can totally include that to the aws-ecs-2* variants. As for variants in general, I only wanted to introduce the `rdma-core` package to the variants that uses kernel-6.1 where EFA drivers are included.

File: sources/settings-migrations/v1.28.0/kernel-sysctl-hugepages/src/main.rs
Comment (ginglis13) [inline sources/settings-migrations/v1.28.0/kernel-sysctl-hugepages/src/main.rs]: I think you want to provide the setting itself here:
```suggestion
        setting: "settings.kernel.sysctl.vm/nr_hugepages",
```

**Comment by ytsssun** [comment]
As per @bcressey , I added the `rdma-core` package in `release` so that it is enabled for aws-* variants. https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/252

Will update this PR once the core-kit PR is merged and getting released.

**Review by bcressey — APPROVED** [review]
LGTM after it's rebased to fix `Release.toml` and `Twoliter.toml`.

**Review by bcressey — APPROVED** [review]

**Review by arnaldo2792 — APPROVED** [review]

**Review by bcressey — APPROVED** [review]

**Pull request merged**
